src/nbl/video/utilities/IUtilities.cpp

```diff
@@ -192,458 +96,48 @@ void IUtilities::updateImageViaStagingBuffer(
         }
         else
         {
-            // Start CmdCopying Regions and Copying Data to m_defaultUploadBuffer
-            uint32_t currentUploadBufferOffset = 0u;
-            uint32_t availableUploadBufferMemory = 0u;
-            auto addToCurrentUploadBufferOffset = [&](uint32_t size) -> bool 
-            {
-                currentUploadBufferOffset += size;
-                currentUploadBufferOffset = core::alignUp(currentUploadBufferOffset, bufferOffsetAlignment);
-                if(currentUploadBufferOffset - localOffset <= uploadBufferSize)
-                {
-                    availableUploadBufferMemory = uploadBufferSize - (currentUploadBufferOffset - localOffset);
-                    return true;
-                }
-                else
-                    return false;
-            };
-
-            // currentUploadBufferOffset = localOffset
-            // currentUploadBufferOffset = alignUp(currentUploadBufferOffset, bufferOffsetAlignment)
-            // availableUploadBufferMemory = uploadBufferSize
-            addToCurrentUploadBufferOffset(localOffset);
+            size_t currentUploadBufferOffset = localOffset;
+            size_t availableUploadBufferMemory = allocationSize;
 
             bool anyTransferRecorded = false;
             core::vector<asset::IImage::SBufferCopy> regionsToCopy;
 
             // Worst case iterations: remaining blocks --> remaining rows --> remaining slices --> full layers
             const uint32_t maxIterations = regions.size() * 4u; 
```

Comment:
can you hoist this and `reserve` the  `regionsToCopy` to be the right size immediately? (then just clear it here)

Reply:
Good idea